pkg/chattr/chattr.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2025 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package chattr
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	// Attribute flags
+	FS_IMMUTABLE_FL = 0x00000010 // Immutable file
+	FS_APPEND_FL    = 0x00000020 // Append-only file
+	// ... other flags as needed
+)
+
+// SetAttr sets the attributes of a file.
+func SetAttr(file *os.File, attrStr string) error {
+	var attr int32
+	switch attrStr {
+	case "+i":
+		attr = FS_IMMUTABLE_FL
+	case "+a":
+		attr = FS_APPEND_FL
+	case "-i":
+		attr = -FS_IMMUTABLE_FL
+	case "-a":
+		attr = -FS_APPEND_FL
+	default:
+		return fmt.Errorf("invalid attribute. Use +i, +a, -i, or -a")
+	}
+
+	currentAttr, err := GetAttr(file)
+	log.Printf("currentAttr: %v %v", currentAttr, err)
+	if err != nil {
+		return err
+	}
+
+	if attr > 0 {
```

Comment:
I think it would be clearer to construct two values `setBits` which are the bits you want to have set, and `clearBits` which are the bits you want to have cleared.

`currentAttr = (currentAttr &^ clearBits) | setBits`